cms/static/js/spec/views/pages/course_outline_spec.js

```diff
@@ -308,6 +323,28 @@ define(["jquery", "js/common_helpers/ajax_helpers", "js/views/utils/view_utils",
                     outlinePage.$('.nav-actions .button-toggle-expand-collapse .expand-all').click();
                     verifyItemsExpanded('section', true);
                 });
+
+                it('can start reindex of a course - respond success', function() {
+                    createCourseOutlinePage(this, mockSingleSectionCourseJSON);
+                    var reindexSpy = spyOn(outlinePage.outlineView, 'startReIndex').andCallThrough();
+                    var successSpy = spyOn(outlinePage.outlineView, 'onIndexSuccess').andCallThrough();
+                    var reindexButton = outlinePage.$('.button.re-index');
+                    reindexButton.trigger('click');
+                    AjaxHelpers.expectJsonRequest(requests, 'GET', '/course_index/5');
+                    AjaxHelpers.respondWithJson(requests, createMockIndexJSON(true));
+                    expect(reindexSpy).toHaveBeenCalled();
+                    expect(successSpy).toHaveBeenCalled();
+                });
+
+                it('can start reindex of a course - respond fail', function() {
+                    createCourseOutlinePage(this, mockSingleSectionCourseJSON);
+                    var reindexSpy = spyOn(outlinePage.outlineView, 'startReIndex').andCallThrough();
+                    var reindexButton = outlinePage.$('.button.re-index');
+                    reindexButton.trigger('click');
+                    AjaxHelpers.expectJsonRequest(requests, 'GET', '/course_index/5');
+                    AjaxHelpers.respondWithJson(requests, createMockIndexJSON(false));
+                    expect(reindexSpy).toHaveBeenCalled();
```

Comment:
Oh wait, you're mocking the response, not getting an actual one. Never mind, sorry.